waspc/data/Generator/templates/server/src/core/auth.js

```diff
@@ -69,5 +70,27 @@ export const verifyPassword = async (hashedPassword, password) => {
   }
 }
 
-export default auth
+// NOTE(shayne): This function helps us handle external auth users.
+// We have two security-related scenarios to consider:
+// 1) If the user is new and comes via external auth first, we simply create
+// their account with a random password. They cannot use `emailAndPassword`
+// to log with this email in until we add password reset functionality.
+// 2) If a user already exists from `emailAndPassword`, we cannot be
+// sure it really belongs to them since we do not yet do email validation.
+// Therefore, we *also* reset the password when we find an existing user, just
+// in case it was someone else who created that account. We do not want them to
+// still have access.
+// Upsert solves for both of these cases (always randomize password) cleanly and efficiently.
```

Comment:
This comments explains where is this function used in the rest of code and why.
I would suggest modify the comment to exclusively comment what the function does, without knowing how it is used in other places, and then in those other places you can explain why you are calling it there.
I am suggesting this due to this function being really standalone. Also, in those other places, if you remove the usage of this function, you might not remember to update this comment. And at the end, it is better for function not to assume how it is used, if possible, as to reduce coupling of any kind.

Reply:
True, that is a good point. I can pull the bulk of this to the use and just document the functionality here.